src/arrayMethodJoin.js

```diff
@@ -4,8 +4,25 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let resultStr = '';
+    let separatorItem;
+
+    if (separator === null) {
+      separatorItem = 'null';
+    } else {
+      separatorItem = separator.toString();
```

Comment:
it looks like you don`t need this if/else condition, because you do the same - you transform your separator to string in both cases;
-you can transform it either in declaration let separatorItem = ' ' + separator
- or you can  + it to you result string, but separatly in new line:
resultStr += this[i];
resultStr += separatorItem; - in this case do do not even need to assaign a variable for separator, just use income separator;
-or other method (like interpolation)
-or leave only first if (for null)